### DISCUSSION TITLE
Is there a way to decrypt downloaded ipxe scripts?

### DISCUSSION THREAD

**Discussion by braindevices** [ipxe/ipxe#468] [General]
I have some secrets in some ipxe scripts. And we fetch those files from a http server which is accessible from internet. I wonder is there a way to decrypt the script or secret in iPXE, apparently we do not want them be plain text on the internet?

**Comment by NiKiZe** [comment]
essentially no, but this is not really a limitation of iPXE itself but rather how "secrets" work.

First you could run some serverside scripting language that only makes the secrets/script available if a secret is sent in.
This secret could be embedded in the ipxe binaries by using embedded script, or even by the DHCP server configuration.

Normally you could do encryption with private/public key pairs, but you would still need to have the private part on the iPXE client - but again as above you could do this in the iPXE binary (embedding a client certifcate) and then use HTTPS to only give out the information to the clients that are allowed to receive it.

So to summarize, try to read information on https client certificates, and use a http server or scripting language where you can use "passwords" to restrict access.
